subsystem/host_test.go

```diff
@@ -477,8 +477,31 @@ var _ = Describe("Host tests", func() {
 			},
 		})
 		Expect(err).To(HaveOccurred())
+		Expect(err).Should(BeAssignableToTypeOf(installer.NewRegisterHostUnauthorized()))
 
 		err = wiremock.DeleteStub(wrongTokenStubID)
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("register_failed_sso", func() {
+		if !Options.EnableAuth {
```

Comment:
What's the purpose? Does it mean that this functionality won't be tested if auth is disabled? I'd probably write two tests then - one with auth disabled and the other with forced enabled.

Reply:
RegisterHost is covered in multiple tests with `registerHost` https://github.com/openshift/assisted-service/blob/master/subsystem/utils_test.go#L29
This specific test is relevant only when Auth is enable, which is by the way the default in sub-system tests unless running `on-prem`.